filter.go

```diff
@@ -22,6 +23,8 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective("filter", parseCaddyfile)
 }
 
+var paramReplacementPattern = regexp.MustCompile("\\{[a-zA-Z0-9_\\-.]+}")
```

Comment:
Can you make the `+` non-greedy by changing it to `+?`? I would expect it to be faster if we use non-greedy matches

Reply:
According to some online blog posts, that should not change anything, as the group only allows a limited set of characters. But I will change it, if you feel more comfortable with it.